hive/src/main/java/org/elasticsearch/hadoop/hive/EsStoragePredicateHandler.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.hive;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hive.ql.plan.ExprNodeDesc;
+import org.apache.hadoop.hive.serde2.Deserializer;
+import org.apache.hadoop.mapred.JobConf;
+import org.elasticsearch.hadoop.cfg.HadoopSettingsManager;
+import org.elasticsearch.hadoop.cfg.Settings;
+import org.elasticsearch.hadoop.hive.pushdown.HiveTreeBuilder;
+import org.elasticsearch.hadoop.hive.pushdown.PredicateHandler;
+import org.elasticsearch.hadoop.hive.pushdown.SargableParser;
+import org.elasticsearch.hadoop.hive.pushdown.Utils;
+import org.elasticsearch.hadoop.hive.pushdown.function.Pair;
+import org.elasticsearch.hadoop.hive.pushdown.node.Node;
+import org.elasticsearch.hadoop.hive.pushdown.node.OpNode;
+import org.elasticsearch.hadoop.hive.pushdown.parse.EsQueryParser;
+import org.elasticsearch.hadoop.hive.pushdown.parse.EsTreeParser;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.JsonObj;
+import org.elasticsearch.hadoop.util.SettingsUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.elasticsearch.hadoop.cfg.ConfigurationOptions.ES_QUERY;
+
+
+/**
+ * EsStoragePredicateHandler is an entrance class for implementing pushdown optimization.
+ */
+public class EsStoragePredicateHandler extends PredicateHandler<JsonObj> {
```

Comment:
Because our company needs to connect to other data sources, such as hbase. Current, It has been changed to one class, and del 'PredicateHandler' class .